functions/database/database.js

```diff
@@ -0,0 +1,316 @@
+const admin = require("firebase-admin");
+const db = admin.firestore();
+const {Timestamp} = require("firebase-admin/firestore");
+const RELEASE_STATES = require("../utils/releaseStates.js");
+const {
+  validateNewReleaseStructure,
+  validateRelease,
+} = require("../validation/validation.js");
+const {parseCommitTitleFromMessage} = require("../utils/utils.js");
+const {REPO_URL} = require("../github/github.js");
+
+/**
+ * Retrieve the release ID of the Firestore release document
+ * with a given release name.
+ *
+ * @param {string} releaseName The release name of the release
+ * document to fetch.
+ * @throws {Error} If there is not exactly one release with the given name.
+ * @return {Promise<string>} A promise that resolves to the release ID.
+ */
+async function getReleaseID(releaseName) {
+  const releaseSnapshot = await db.collection("releases")
+      .where("releaseName", "==", releaseName)
+      .get();
+
+  if (releaseSnapshot.empty) {
+    throw new Error(`No release found with name: ${releaseName}`);
+  }
+
+  // There should only be one release with a given name
+  if (releaseSnapshot.size > 1) {
+    throw new Error(`${releaseSnapshot.size} releases found with name:
+     ${releaseName}`);
+  }
+
+  const releaseId = releaseSnapshot.docs[0].id;
+
+  return releaseId;
+}
+
+/**
+ * Retrieves release data from the database given a release ID.
+ *
+ * @param {string} releaseId The ID of the release to fetch.
+ * @return {Promise<Object>} A promise that resolves to the release data.
```

Comment:
Missing `@throws {Error}`

Reply:
Resolved in https://github.com/firebase/firebase-release-dashboard/pull/8/commits/a32e049b16404c7818f488231642a11b343b3068